lib/datadog/statsd.rb

```diff
@@ -90,15 +95,20 @@ def self.VERSION
     # @param [Integer] port your statsd port
     # @option opts [String] :namespace set a namespace to be prepended to every metric name
     # @option opts [Array<String>] :tags tags to be added to every metric
-    def initialize(host = DEFAULT_HOST, port = DEFAULT_PORT, opts = {}, max_buffer_size=50)
+    # @option opts [Integer] :max_buffer_size max messages to buffer
+    # TODO: remove bad_old_max_buffer_size on next major bump
+    def initialize(host = DEFAULT_HOST, port = DEFAULT_PORT, opts = {}, bad_old_max_buffer_size=50)
       self.host, self.port = host, port
       @socket_path = opts[:socket_path]
       @prefix = nil
       @socket = connect_to_socket if @socket_path.nil?
       self.namespace = opts[:namespace]
       self.tags = opts[:tags]
-      @buffer = Array.new
-      self.max_buffer_size = max_buffer_size
+
+      # buffering
+      @max_buffer_size = opts[:max_buffer_size] || bad_old_max_buffer_size
+      @buffer = String.new
```

Comment:
not using `""` to avoid it being frozen when we switch to frozen strings

Reply:
Can you add this in a comment?

Reply:
added